Reneiski_Homework1/Reneiski_Homework1/Program.cs

```diff
@@ -1,28 +1,279 @@
+            //значения так же заданы не по алфавиту. Напишите метод, который совершает операции сортировки двух списков, списка int – по возрастанию,
+            //списка string – по убыванию. Затем данный метод объединяет списки в словарь.Полученный словать выведите на консоль
+
+            //DictionarySecondAction();
+
+            static void DictionarySecondAction()
+            {
+                List<int> ints = new List<int>() { 1, 3, 5, 7, 9, 2, 4, 6, 8, 10 };
+                List<string> strings = new List<string>() { "a", "c", "e", "g", "i", "b", "d", "f", "h", "j" };
+
+                var orderedInts = ints.OrderBy(p => p).ToList();
+
+                foreach (var item in orderedInts)
+                {
+                    Console.WriteLine(item);
+                }
+
+                var oderedStrings = strings.OrderByDescending(p => p).ToList();
+
+                foreach (var item in oderedStrings)
+                {
+                    Console.WriteLine(item);
+                }
+
+                var merged = orderedInts.Zip(oderedStrings, (key, value) => (key, value)).ToDictionary(p => p.key, p => p.value);
+
+                foreach (var item in merged)
+                {
+                    Console.WriteLine(item);
+                }
+            }
+
+            //Создайте класс City, где есть поля int население, double площадь.
+            //Создайте словарь, где Key – название города, а Value – сооттветствующий названию города объект типа City.
+            //Создайте 5 элементов для словаря. 
+            //  a.Отсортируйте словарь по площади города и выведите на консоль
+            //  b.Остортируйте словарь по населению в обратном порядке и выведите на консоль
+            //  c.Подсчитайте общее население все городов и выведите на консоль
+
+            var cities = new Dictionary<string, City>
+            {
+                {"Minsk", new City() { Population = 1975000, Square = 348.8 } },
+                {"Gomel", new City() {Population = 508839, Square = 139.8} },
+                {"Grodno", new City() {Population = 368710, Square = 142.1} },
+                {"Vitebsk", new City() {Population = 362466, Square = 124.5} },
+                {"Brest", new City() {Population = 343985, Square = 146.1} }
+            };
+
+            //PrintCitiesDict(cities);
+
+            static void PrintCitiesDict(Dictionary<string, City> cities)
+            {
+                foreach (var item in cities.Keys)
+                {
+                    Console.WriteLine($"{item}, {cities[item].Population}, {cities[item].Square}");
+                }
+            }
+
+            Console.WriteLine();
+            Console.WriteLine("Sort dictionary by Square");
+
+            var sortedBySquare = cities.OrderBy(p => p.Value.Square);
+
+            var printSortedBySquare = sortedBySquare.ToDictionary(p => p.Key, p => p.Value);
```

Comment:
I think you can remove this variable and just apply ToDictionary method to cities.OrderBy(p => p.Value.Square) expression

Reply:
Fixed